agent/util-scripts/pbench-list-tools

```diff
@@ -19,24 +19,28 @@ with_options=0
 
 # Process options and arguments
 
-opts=$(getopt -q -o tn:og: --longoptions "with-options,name:,group:" -n "getopt.sh" -- "$@");
-if [ $? -ne 0 ]; then
+function usage() {
 	printf "\n"
-	printf "$script_name: you specified an invalid option\n\n"
 	printf "The following options are available (use neither, one, or the other, but not both):\n\n"
 	printf -- "\t-n str --name=str, list the perftool groups which str is used\n"
 	printf "\n"
 	printf -- "\t-g str --group=str, list the tools used in this str\n"
 	printf "\n"
 	printf -- "\t-o --with-options, list the options with each tool\n"
+}
 
-
+opts=$(getopt -q -o hon:g: --longoptions "help,with-options,name:,group:" -n "getopt.sh" -- "$@");
+if [ $? -ne 0 ]; then
+	printf -- "\n${*}\n" >&2
+	printf -- "${script_name}: you specified an invalid option\n\n" >&2
+	usage >&2
 	exit 1
```

Comment:
Is there a unit test which exercises this code?

Reply:
It was exercised when `help` was an invalid option, but no longer :-)